app/scripts/models/account.js

```diff
@@ -196,21 +196,30 @@ define(function (require, exports, module) {
         return p.reject(AuthErrors.toError('INVALID_TOKEN'));
       }
 
-      // assertions live for 25 years, they can be cached and reused while
-      // this browser tab is open.
       var existingAssertionPromise = this._assertionPromises[sessionToken];
 
-      if (existingAssertionPromise) {
+      if (this._isAssertionValid(existingAssertionPromise)) {
         return existingAssertionPromise;
       }
 
       var assertionPromise = this._assertion.generate(sessionToken);
+      assertionPromise.__createdAt = Date.now();
 
       this._assertionPromises[sessionToken] = assertionPromise;
 
       return assertionPromise;
     },
 
+    _isAssertionValid (assertionPromise) {
+      // assertions live for about 6 hours.
+      // reuse the same assertion if used in the past hour
```

Comment:
"used in the past hour" or "created in the past hour"?

Reply:
true!